### PR TITLE
Add Upload-Stream header

### PR DESCRIPTION
**Pull request by vayam**

### CODE REVIEW

File: protocol.md [hunk 1]
```diff
@@ -397,14 +395,23 @@ This extension defines how to upload finite streams of data that have an
...
 
+#### Headers
+
+##### Upload-Stream
+
+The `Upload-Stream: 1` MUST be included with the file creation request.
+
+##### Upload-Length
+
+The `Upload-Length` MUST be set only once with the `PATCH` request.
```
Comment (kvz) [inline protocol.md]: I think the client can sent as many as it wants, but the server only accepts the first one? That will make implementations with parallel running workers easier on the client side

Comment (vayam) [inline protocol.md]: @kvz  oops I dint realize I was changing the meaning. putting back to  `Upload-Length` once set MUST NOT be changed.

File: protocol.md [hunk 2]
```diff
@@ -249,9 +249,7 @@ core protocol for performing the actual upload.
 ##### Upload-Length
 
 The `Upload-Length` header indicates the final size of the new upload in bytes.
-This way a Server will implicitly know when a file has completed uploading. The
```
Comment (Acconut) [inline protocol.md]: > This way a Server will implicitly know when a file has completed uploading.
I think this sentence is important enough to stay.

File: protocol.md [hunk 3]
```diff
@@ -265,9 +263,9 @@ key SHOULD be ASCII encoded and the value MUST be Base64 encoded.
 ##### POST
 
 Clients MUST use a `POST` against a well known upload creation URL to request the
-creation of a new file resource. The request MUST include an `Upload-Length`
-header. If the [Stream](#stream) extension is used to upload a file of unknown
-size the header `Upload-Length: streaming` MUST be included.
+creation of a new file resource. The request MUST include `Upload-Length`
```
Comment (Acconut) [inline protocol.md]: include *the* `Upload-Length` header

File: protocol.md [hunk 4]
```diff
@@ -265,9 +263,9 @@ key SHOULD be ASCII encoded and the value MUST be Base64 encoded.
...
+header when upload size is known. The request MUST include `Upload-Stream: 1` when
+upload size is unknown. The headers included here SHALL be dictated by dependent extensions.
```
Comment (Acconut) [inline protocol.md]: Either move the mention of `Upload-Stream` to the Stream extension or link to it.

File: protocol.md [hunk 5]
```diff
@@ -397,14 +395,22 @@ This extension defines how to upload finite streams of data that have an
...
-entire upload is known it MUST be included as the `Upload-Length` header's value
-in the next `PATCH` request. Once the upload's length has been set it MUST NOT
-be changed.
+`Upload-Stream` header MUST be set to 1. Once the total size of the entire 
+upload is known `Upload-Length` MUST be set in the next `PATCH` request. 
 
 In order to indicate that this extension is supported by the Server it MUST
 include the `stream` element in the `Tus-Extension` header.
 
+#### Headers
```
Comment (Acconut) [inline protocol.md]: The Headers section contains definitions of all new headers defined by this extension. Please remove `Upload-Length` from this section and move the note about the immutability of the upload's length somewhere else. In addition please add a sentence to `Upload-Stream` that it's usage to indicate the usage of the Stream extension.

File: protocol.md [hunk 6]
```diff
@@ -397,14 +395,22 @@ This extension defines how to upload finite streams of data that have an
 unknown length at the beginning of the upload.
 
 If the [Creation](#creation) extension is used to initiate a new upload the
-`Upload-Length` header MUST be set to the string `streaming`. Once the total size of the
-entire upload is known it MUST be included as the `Upload-Length` header's value
-in the next `PATCH` request. Once the upload's length has been set it MUST NOT
-be changed.
+`Upload-Stream` header MUST be set to 1. Once the total size of the entire 
+upload is known `Upload-Length` MUST be set in the next `PATCH` request.
```
Comment (Acconut) [inline protocol.md]: Add that the `Upload-Length` header must not be contained in the `POST` request.

File: protocol.md (hunk 2 above)
Comment (vayam) [inline protocol.md]: I disagree. But I will add it back.

**Comment by vayam** [comment]
We decided to drop `Upload-Stream` header altogether.

**Pull request closed without merging**